packages/studio-base/src/providers/LayoutManagerProvider.test.tsx

```diff
@@ -0,0 +1,217 @@
+/** @jest-environment jsdom */
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import { render, waitFor } from "@testing-library/react";
+import { useNetworkState } from "react-use";
+
+import { useVisibilityState } from "@foxglove/hooks";
+import { LayoutData } from "@foxglove/studio-base/context/CurrentLayoutContext";
+import { useLayoutStorage } from "@foxglove/studio-base/context/LayoutStorageContext";
+import { useRemoteLayoutStorage } from "@foxglove/studio-base/context/RemoteLayoutStorageContext";
+import LayoutManagerProvider from "@foxglove/studio-base/providers/LayoutManagerProvider";
+import { LayoutLoader } from "@foxglove/studio-base/services/ILayoutLoader";
+import MockLayoutManager from "@foxglove/studio-base/services/LayoutManager/MockLayoutManager";
+
+// Mock dependencies
+jest.mock("react-use");
+jest.mock("@foxglove/hooks");
+jest.mock("@foxglove/studio-base/context/LayoutStorageContext");
+jest.mock("@foxglove/studio-base/context/RemoteLayoutStorageContext");
+
+const mockLayoutManager = new MockLayoutManager();
```

Comment:
Can you please add those test scenarios? 

- When filter is rejected and the following message is being shown:

 ```
  loaderResults.filter(isRejected).forEach((result) => {
          log.error(`Failed to fetch layouts from loader: ${result.reason}`);
        });
```
When loading layouts fail and the following message is being shown:

`} catch (err) {
        log.error("Loading default layouts failed:", err);
      }`

Reply:
Good point. I create a test for each of this 3 topics and one to handle partial success and error, meaning one `fetchLayouts` having error and one `saveLayouts`, so it should save the correct ones and log the wrong ones, as expected.